dm/pkg/etcdutil/etcdutil.go

```diff
@@ -129,6 +145,18 @@ func DoOpsInOneCmpsTxnWithRetry(cli *clientv3.Client, cmps []clientv3.Cmp, opsTh
 	return resp, resp.Header.Revision, nil
 }
 
+// DoEtcdOpsWithRepeatableRetry do etcd operations function with repeatable retry.
+func DoEtcdOpsWithRepeatableRetry(cli *clientv3.Client, operateFunc func() error) error {
+	ctx, cancel := context.WithTimeout(cli.Ctx(), DefaultRequestTimeout)
+	defer cancel()
+	tctx := tcontext.NewContext(ctx, log.L())
+
+	_, _, err := etcdDefaultTxnStrategy.Apply(tctx, etcdSafeTxnRetryParam, func(t *tcontext.Context) (ret interface{}, err error) {
+		return nil, operateFunc()
+	})
```

Comment:
I thought about this carefully. `operateFunc` and `ops...` parameters are doing almost the thing - to do some etcd operations. If we add both two parameters in one function it might be confusing for users to decide which one to assign. Maybe keeping this for now?

Reply:
i mean abstract a func like below which is op neutral, and other public code can call it:
```golang
func doWithRetry(cli *clientv3.Client, opFunc func(t *tcontext.Context) (ret interface{}, err error)) (*clientv3.TxnResponse, int64, error) {
	ctx, cancel := context.WithTimeout(cli.Ctx(), DefaultRequestTimeout)
	defer cancel()
	tctx := tcontext.NewContext(ctx, log.L())

	ret, _, err := etcdDefaultTxnStrategy.Apply(tctx, etcdDefaultTxnRetryParam, opFunc)
	if err != nil {
		return nil, 0, err
	}
	resp := ret.(*clientv3.TxnResponse)
	return resp, resp.Header.Revision, nil
}
```